src/Frontend/components/app/layout/Application.js

```diff
@@ -24,7 +25,9 @@ import AppMenu from './AppMenu';
 const Application = props => {
   return (
     <div className="container">
-      <AppMenu config={menu} />
+      <Provider store={store}>
```

Comment:
had a look at this again.
I think that the `connect()` builds a higher-order-component which injects all the needed information into the actual `AppMenu` component.
The `Provider` simply creates a context with the store, but each sub-component (in any depth) can access the context, so it should be possible to remove the `<Provider></Provider>` thing here and simply leave the `AppMenu` at this place.